AABB_tree/benchmark/AABB_tree/CMakeLists.txt

```diff
@@ -23,9 +23,18 @@ if ( NOT Boost_FOUND )
   return()  
 endif()
 
+# google benchmark
+find_package(benchmark REQUIRED)
+if ( NOT benchmark_FOUND )
+  message(STATUS "This project requires Google benchmarks, and will not be compiled.")
+endif()
```

Comment:
[Doc of `REQUIRED`](https://cmake.org/cmake/help/latest/command/find_package.html):
>  The `REQUIRED` option stops processing with an error message if the package cannot be found.

That means that, if `benchmark` is not found, then the next lines will not be processed anyway. You can remove the `if(NOT benchmark_FOUND)`.